libraries/AP_Notify/NeoPixel_Oreo.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "NeoPixel_Oreo.h"
+#include "NeoPixel.h"
+
+extern const AP_HAL::HAL& hal;
+
+NeoPixel_Oreo::NeoPixel_Oreo(uint8_t theme) :
+    NotifyDevice()
+{
+    oreo = new OreoLED_I2C(0,theme);
```

Comment:
better to break out the needed theme functions from oreoled instead of instantiating an i2c driver

Reply:
I tried this and it's going to be a major change with huge risk of breaking OREO for Solo. I'd reaaaaally like to not do this on this PR.

Reply:
I tried making OreoLED_I2C statically assigned in NeoPixel_Oreo.h but since it's touching the HAL I2C drivers it complains about the deconstructor

```

In file included from ../../libraries/AP_HAL/SPIDevice.h:23:0,
                 from ../../libraries/AP_HAL/HAL.h:11,
                 from ../../libraries/AP_HAL/AP_HAL_Main.h:19,
                 from ../../libraries/AP_HAL/AP_HAL.h:8,
                 from ../../libraries/AP_Notify/NeoPixel_Oreo.h:17,
                 from ../../libraries/AP_Notify/NeoPixel_Oreo.cpp:16:
../../libraries/AP_HAL/utility/OwnPtr.h: In instantiation of ‘AP_HAL::OwnPtr<T>::~OwnPtr() [with T = AP_HAL::I2CDevice]’:
../../libraries/AP_Notify/OreoLED_I2C.h:28:7:   required from here
../../libraries/AP_HAL/utility/OwnPtr.h:74:17: warning: possible problem detected in invocation of delete operator: [-Wdelete-incomplete]
     ~OwnPtr() { delete _ptr; }
                 ^~~~~~
../../libraries/AP_HAL/utility/OwnPtr.h:74:24: warning: invalid use of incomplete type ‘class AP_HAL::I2CDevice’
     ~OwnPtr() { delete _ptr; }
                        ^~~~
In file included from ../../libraries/AP_HAL/AP_HAL.h:5:0,
                 from ../../libraries/AP_Notify/NeoPixel_Oreo.h:17,
                 from ../../libraries/AP_Notify/NeoPixel_Oreo.cpp:16:
../../libraries/AP_HAL/AP_HAL_Namespace.h:13:11: note: forward declaration of ‘class AP_HAL::I2CDevice’
     class I2CDevice;
           ^~~~~~~~~
In file included from ../../libraries/AP_HAL/SPIDevice.h:23:0,
                 from ../../libraries/AP_HAL/HAL.h:11,
                 from ../../libraries/AP_HAL/AP_HAL_Main.h:19,
                 from ../../libraries/AP_HAL/AP_HAL.h:8,
                 from ../../libraries/AP_Notify/NeoPixel_Oreo.h:17,
                 from ../../libraries/AP_Notify/NeoPixel_Oreo.cpp:16:
../../libraries/AP_HAL/utility/OwnPtr.h:74:17: note: neither the destructor nor the class-specific operator delete will be called, even if they are declared when the class is defined
     ~OwnPtr() { delete _ptr; }
                 ^~~~~~

```

Since we aren't short on RAM I think we should leave it like this at least for now. The "parent" class of NeoPixel_Oreo only has 3 other variables (two primitive plus a mutex) in the class so the memory footprint is basically the same as just instantiating the original driver